src/flow.jl

```diff
@@ -207,10 +207,10 @@ function update(sf::SurfaceFlowLand, network, frac_toriver)
     @. sf.qlat = sf.inwater / sf.dl
 
     # Convert threshold to volume
-    threshold_volume = sf.h_thresh .* sf.width .* sf.dl
+    threshold_volume = sf.h_thresh .* area
```

Comment:
Why the use of `area` here? It's confusing as it deviates from the overland flow dimensions `width` and `dl`. For river cells `area` is higher than `sf.width .* sf.dl`, for land cells it has the same value. I would suggest to use the overland flow dimensions for the computation of the `threshold_volume`, and the `pond_volume_current`. 

What most probably is not correct yet when using the overland flow dimensions, is the amount of surface water that can infiltrate from river cells. I think this amount should be scaled by the land fraction of these cells. We have a similar approach for evaporation and direct runoff by using `waterfrac`.